src/core/network/client.ts

```diff
@@ -342,8 +347,10 @@ class ApiClient {
     }
 
     debug('HTTP call', method, payload)
-    const buildConfig = includesMedia(payload)
-      ? buildFormDataConfig({ method, ...payload }, options.agent)
+    // I hate these `?? {}`, but I don't know how to
+    // make empty payload `{}` instead of `undefined`
+    const buildConfig = includesMedia(payload ?? {})
+      ? buildFormDataConfig({ method, ...(payload ?? {}) }, options.agent)
```

Comment:
@KnorpelSenf?

Reply:
Regarding line 352, I suggest you simply allow `undefined` in the method signature of `includesMedia` and in turn go with `return payload !== undefined && Object.values(payload).some(` …

The second one in line 353 is a little problematic to avoid. I suggest you either
- leave the `?? {}` in there and just avoid reading the code ever again, or maybe
- do a QND like this:
   ```ts
    const m: Record<string, unknown> = { method }
    const buildConfig = includesMedia(payload)
      ? buildFormDataConfig({ ...m, ...payload }, options.agent)
      : buildJSONConfig(payload)
   ```
I can't really come up with a nice and clean solution here, but it might just be missing something.

Reply:
To be clear, I was wondering whether it would be possible to make for example `Opts<'getChat'>` return `{}` instead of `undefined`.

Reply:
I see. I'll have a look if I find a way to do this, I agree it would make things more comfortable.

Reply:
@wojpawlik check v2.2.0

Reply:
`Opts<'getChat'>` isn't `undefined` btw, but other methods are, I'm assuming you're referring to them.

Reply:
I must've meant `Opts<'getMe'>`, I need a break 😅

Sadly, v2.2.0 doesn't help. I did try writing my own `Opts` before I asked you, and no matter what I did, I couldn't convince TS that the value can't be `undefined`.

Reply:
You're right, sorry. I cloned your fork, reproduced and fixed the issue. The problem is that conditional types are transformed to union types on their way. I published this in 2.2.1, could you try that version? Do let me know if you're having further issues, perhaps I make a mistake while fixing it or something.

Reply:
It works now, I'm impressed.
If you have time to spare, it'd be nice to port `Markup` to TS (breaking changes allowed).